### ISSUE TITLE
Lecterns in the player's inventory are rendered differently between Minecraft Vanilla and NeoForge

### ISSUE THREAD

**Issue by Lolothepro** [neoforged/NeoForge#144]
Neo:
![image](https://github.com/neoforged/NeoForge/assets/68275908/b8aff9d1-cae1-4103-b64e-ca577586d1e9)
Vanilla:
![image](https://github.com/neoforged/NeoForge/assets/68275908/6c969ed9-9d46-4e3d-bc8a-637ef5244961)

![image](https://github.com/neoforged/NeoForge/assets/68275908/e1e5d353-b5a8-4166-83fb-91cacbef68e8)
![image](https://github.com/neoforged/NeoForge/assets/68275908/87154e13-7ff1-40a0-92d1-d7c00b9fcdd3)

**Comment by Technici4n** [comment]
That's probably a consequence of different normal orientation

**Comment by XFactHD** [comment]
My best guess is that this is caused by Forge calculating proper normals from the quad's vertices while vanilla uses the normals of the closest `Direction`: https://github.com/neoforged/NeoForge/blob/1.20.x/patches/minecraft/net/minecraft/client/renderer/block/model/FaceBakery.java.patch#L8.

In the discussion on Discord it was suggested to gate this behind a config option that defaults to true in case someone doesn't like this visual change.

**Comment by Technici4n** [comment]
Disabling it will break many modded models. I'd like to find a proper solution as part of the chunk rendering rewrite.

**Comment by XFactHD** [comment]
This is not chunk rendering though, it's item rendering in a GUI.
Apart from that, what Forge does in this case is technically the proper solution, it just seems like some people don't want this kind of fix.

**Comment by Technici4n** [comment]
It's deeply related to how normals are used for chunk rendering. "Chunk rendering rewrite" also covers baked models in my mind.

**Comment by Shadows-of-Fire** [comment]
I would argue this is one of those things we should just mark as `wontfix` and leave it at that.  We shouldn't be special casing things out of the proper solution just because vanilla can't properly compute normals.

**Comment by KnightMiner** [comment]
To be clear, which lecture in which in the closeup screenshots? I think the order was swapped and Neo is the darker one?

If vanilla is the brighter one, I could see a case for keeping that behavior on item models as it leads to a more visible item, perhaps as a model JSON flag that defaults models under `minecraft` to vanilla behavior and other models to the current forge behavior?

**Comment by XFactHD** [comment]
Neo is the top one with the brighter top face.

**Comment by Technici4n** [comment]
Vanilla should be the one with the brighter top face

**Comment by PlatinPython** [comment]
Not in the larger pictures tho. There vanilla is brighter. The order of the cropped images is different from the large ones.

**Comment by XFactHD** [comment]
I re-checked in-game to be sure and the brighter one is indeed Vanilla, so ignore my earlier comment.

Vanilla:
![2023-09-25_19 21 48](https://github.com/neoforged/NeoForge/assets/11262040/1cda2f29-33af-4190-93ea-e039aa0e7e93)

Neo:
![2023-09-25_19 22 04](https://github.com/neoforged/NeoForge/assets/11262040/080c5bab-b866-443b-a367-59fc1da5a374)

**Comment by Technici4n** [comment]
@XFactHD @embeddedt thoughts on this issue? 🤔

**Comment by XFactHD** [comment]
I don't think this is enough of a difference to make a big deal out of it (i.e. justify removing or guarding the baked normal computation in `FaceBakery`).

**Comment by embeddedt** [comment]
Agreed

**Comment by Technici4n** [comment]
Okay, marking as wontfix then as this is pretty minor.
